File: solutions/027/027.py
"""
Project Euler - Problem Solution 027
Problem Title - Quadratic primes
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

#Currently takes ~9s

def is_prime(n): 
  return n > 1 and all(n % p for p in range(2, int(n**0.5)))

def quadratic_formula(n, a, b):
  return n**2 + a*n + b

def consecutive_primes(a, b):
  n = 0
  while True:
    num = quadratic_formula(n, a, b)
    if is_prime(num):
      n += 1
    else:
      break
  return n

def quadratic_primes():
  most_consecutive_primes = 0
  best_coefficients = (0, 0)
  for a in range(-999, 1000):
    for b in range(-1000, 1001):
      num_primes = consecutive_primes(a, b)
      if num_primes > most_consecutive_primes:
        most_consecutive_primes = num_primes
        best_coefficients = (a, b)
  return best_coefficients[0] * best_coefficients[1]

if __name__ == "__main__":
	print(quadratic_primes())

File: solutions/027/027.js
/**
 * Project Euler - Problem Solution 027
 * Problem Title - Quadratic primes
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */

//Currently takes ~0.28s

const isPrime = (n) => {
  for(let i = 2; i <= n**0.5; i++)
    if(n % i === 0) return false; 
  return n > 1;
};

const quadraticFormula = (n, a, b) => {
  return n**2 + a*n + b;
};

const consecutivePrimes = (a, b) => {
  let n = 0;
  while(true) {
    const num = quadraticFormula(n, a, b);
    if(isPrime(num)) {
      n++;
    } else {
      break;
    }
  }
  return n;
};

const quadraticPrimes = () => {
  let mostConsecutivePrimes = 0;
  let bestCoefficients = [0, 0];
  for(let a = -999; a < 1000; a++) {
    for(let b = -1000; b < 1001; b++) {
      const numPrimes = consecutivePrimes(a, b);
      if(numPrimes > mostConsecutivePrimes) {
        mostConsecutivePrimes = numPrimes;
        bestCoefficients = [a, b];
      }
    }
  }
  return bestCoefficients[0] * bestCoefficients[1];
};

console.log(quadraticPrimes());